packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp

```diff
@@ -310,15 +310,15 @@ TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL(Ifpack2OverlappingRowMatrix, getLocalDiag, Sca
 
   Array<scalar_type> val(3);
   Array<global_ordinal_type> ind(3);
-  scalar_type zero=0;
-  scalar_type one=1;
-  val[0] = as<scalar_type>(-1);
-  val[1] = as<scalar_type>(2);
-  val[2] = as<scalar_type>(-1);
+  scalar_type zero=0.0;
+  scalar_type one=1.0;
+  val[0] = static_cast<scalar_type>(-1.0);
```

Comment:
I'm curious, why is `static_cast` preferred over `Teuchos::as`?

Reply:
@jhux2 I'm trying to move to more standard language features as much as possible, partly in hopes of saving build time and partly because they are easier for novice developers to understand.  Also, the main value of `Teuchos::as` is that it may check for overflow in a debug build.  If the cast can't overflow, then there's no reason to use `Teuchos::as`.

Reply:
Thanks, that's good to know.